docs/LoaderApplicationInterface.md

```diff
@@ -0,0 +1,663 @@
+Thus, the ABI number is not encoded in the base of the library filename as on
+Windows.
+On Linux an application wanting to link to the latest Vulkan ABI version would
+just link to the name vulkan (`libvulkan.so`).  A specific Vulkan ABI version
+can also be linked to by applications (e.g. `libvulkan.so.1`).
+On macOS, the libraries are `libvulkan.dylib` and `libvulkan.1.dylib`.
+
+
+## Application Layer Usage
+
+Applications desiring Vulkan functionality beyond what the core API offers may
+use various layers or extensions.
+A layer cannot introduce new Vulkan core API entry-points to an application
+that are not exposed in Vulkan.h.
+However, layers may offer extensions that introduce new Vulkan commands that
+can be queried through the extension interface.
+
+A common use of layers is for API validation which can be enabled by loading
+the layer during application development, but not loading the layer for
+application release.
+This eliminates the overhead of validating the application's usage of the API,
+something that wasn't available on some previous graphics APIs.
+
+To find out what layers are available to your application, use
+`vkEnumerateInstanceLayerProperties`.
+This will report all layers that have been discovered by the loader.
+The loader looks in various locations to find layers on the system.
+For more information see the
+[Layer discovery](LoaderLayerInterface.md#layer-discovery)
+section in the
+[LoaderLayerInterface.md document](LoaderLayerInterface.md) document.
+
+To enable a layer, or layers, simply pass the name of the layers you wish to
+enable in the `ppEnabledLayerNames` field of the `VkInstanceCreateInfo` during
+a call to `vkCreateInstance`.
+Once done, the layers you have enabled will be active for all Vulkan functions
+using the created `VkInstance`, and any of its child objects.
+
+**NOTE:** Layer ordering is important in several cases since some layers
+interact with each other.
+Be careful when enabling layers as this may be the case.
+See the [Overall Layer Ordering](#overall-layer-ordering) section for more
+information.
+
+The following code section shows how you would go about enabling the
+VK_LAYER_KHRONOS_validation layer.
+
+```
+char *instance_validation_layers[] = {
+    "VK_LAYER_KHRONOS_validation"
+};
+const VkApplicationInfo app = {
+    .sType = VK_STRUCTURE_TYPE_APPLICATION_INFO,
+    .pNext = NULL,
+    .pApplicationName = "TEST_APP",
+    .applicationVersion = 0,
+    .pEngineName = "TEST_ENGINE",
+    .engineVersion = 0,
+    .apiVersion = VK_API_VERSION_1_0,
+};
+VkInstanceCreateInfo inst_info = {
+    .sType = VK_STRUCTURE_TYPE_INSTANCE_CREATE_INFO,
+    .pNext = NULL,
+    .pApplicationInfo = &app,
+    .enabledLayerCount = 1,
+    .ppEnabledLayerNames = (const char *const *)instance_validation_layers,
+    .enabledExtensionCount = 0,
+    .ppEnabledExtensionNames = NULL,
+};
+err = vkCreateInstance(&inst_info, NULL, &demo->inst);
```

Comment:
Possibly check for `VK_ERROR_LAYER_NOT_PRESENT`? Since its a something to think of. Feel free to dismiss, this is not 'demo code' just showcasing how layers are enabled.